src/app/BUILD.gn

```diff
@@ -59,6 +59,7 @@ buildconfig_header("app_buildconfig") {
     "CHIP_CONFIG_ENABLE_EVENTLIST_ATTRIBUTE=${enable_eventlist_attribute}",
     "CHIP_CONFIG_ENABLE_ICD_SERVER=${chip_enable_icd_server}",
     "CHIP_CONFIG_ENABLE_READ_CLIENT=${chip_enable_read_client}",
+    "ICD_REPORT_ON_ENTER_ACTIVE_MODE=${chip_report_on_active_mode}",
```

Comment:
OK, so this adds ICD_REPORT_ON_ENTER_ACTIVE_MODE to AppBuildConfig.h.

But the macro is used in `ReportSchedulerImpl.cpp`, which is not including either AppBuildConfig.h or AppConfig.h.  It should include one of those two, I would think, to pick up the macro.  Right now it _might_ be accidentally picking it up via the headers it does include, but that's really fragile.

Reply:
Added explicit include of AppBuildConfig.h